Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -688,15 +688,21 @@ public Declaration FindMemberEnclosingModule(Declaration callingModule, Declarat
 
         public Declaration FindMemberEnclosingProcedure(Declaration enclosingProcedure, string memberName, DeclarationType memberType, ParserRuleContext onSiteContext = null)
         {
-            if (memberType == DeclarationType.Variable && NameComparer.Equals(enclosingProcedure.IdentifierName, memberName))
-            {
-                return enclosingProcedure;
-            }
             var allMatches = MatchName(memberName);
             var memberMatches = allMatches.Where(m =>
                 m.DeclarationType.HasFlag(memberType)
                 && enclosingProcedure.Equals(m.ParentDeclaration));
-            return memberMatches.FirstOrDefault();
+            if (memberMatches.Any())
+            {
+                return memberMatches.FirstOrDefault();
+            }
+
+            if (memberType == DeclarationType.Variable && NameComparer.Equals(enclosingProcedure.IdentifierName, memberName))
+            {
+                return enclosingProcedure;
+            }
+
+            return null;
```

Comment:
Is returning `null `a fairly frequent expected outcome?  If so, would using the TryGet* paradigm be easier and more descriptive for the clients?  I found 4 references to this method, they all check the result against `null ` (and other conditions).  Seems like the `null` check can be handled here - and the client(s) to check for the other conditions relevant to their purpose(s).

Reply:
I had to add something to avoid `not all code paths return value` error. I believe in practice this is never actually reached. I suppose I could substitute this with a `throw new InvalidOperation` exception or something.

Reply:
IMO the `null` return seems preferable to an exception.